src/test/java/com/puppycrawl/tools/checkstyle/bdd/TestInputViolation.java

```diff
@@ -0,0 +1,73 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.bdd;
+
+import java.util.regex.Pattern;
+
+public final class TestInputViolation {
+
+    /** Pattern to match the symbol: "{". */
+    private static final Pattern OPEN_CURLY_PATTERN = Pattern.compile("\\{");
+
+    /** Parsed violation line number. */
+    private final int violationLine;
+
+    /** Parsed violation message. */
+    private final String violationMessage;
+
+    /**
+     * Creates a new {@code TestInputViolation} instance.
+     *
+     * @param violationLine the violation line number.
+     * @param violationMessage the violation message.
+     */
+    public TestInputViolation(int violationLine, String violationMessage) {
+        this.violationLine = violationLine;
+        this.violationMessage = violationMessage;
+    }
+
+    /**
+     * Gets the violation line number.
+     *
+     * @return the violation line number.
+     */
+    public int getViolationLine() {
+        return violationLine;
+    }
+
+    /**
+     * Gets the violation message.
+     *
+     * @return the violation message.
+     */
+    public String getViolationMessage() {
+        return violationMessage;
+    }
+
+    /**
+     * Creates regex string to match the violation message format.
+     *
+     * @return the regex string.
+     */
+    public String toRegex() {
+        return violationLine + ":\\d*:\\s"
+                + OPEN_CURLY_PATTERN.matcher(violationMessage).replaceAll("\\\\{");
```

Comment:
why you don't use violation line number?

remember
1. for some cases we will have no violation message, it means we should compare only line numbers
2. checkstyle violation could start either with line number or line number + column number, you have to generate regex that will match both
- `10: check violation message`
- `10:24: check violation message`

Reply:
> why you don't use violation line number?

I have used it. Line 70 is ```violationLine + ":\\d*:\\s"```.

Reply:
> for some cases we will have no violation message, it means we should compare only line numbers

for this, we can implement condition in ```toRegex``` method, if message is not available then use ```.*``` to construct regex.

Reply:
> checkstyle violation could start either with line number or line number + column number, you have to generate regex that will match both

this regex I wrote ```":\\d*:\\s"``` takes care of it